capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -349,4 +352,116 @@ else if (key.equals("remove_"+name)) {
     }
     return getById(id, collectionReference, response, genericClass).get(0);
   }
-}
\ No newline at end of file
+
+  /**
+   * This method assures that the given post request is valid, contains the necessary fields, and 
+   * does not contain keys that do not match object fields exactly.
+   * @param {jsonObject} the supplied request body in json form
+   * @param {response} response returned from the call
+   * @param {fields} the Fields (contains name and type) of the class being used  
+   * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
+   * @return boolean true if it is a valid post request, false if not
+   */
+  public static <T extends BaseEntity> boolean postErrorHandler(JsonObject jsonObject, HttpServletResponse response, 
+      Field[] fields, List<String> requiredFields) throws IOException {
+    List<String> list = new ArrayList<>();
+    List<String> fieldNames = Arrays.asList(fields).stream().map(f -> f.getName()).collect(Collectors.toList());
+    list.addAll(jsonObject.keySet());
+    list.retainAll(fieldNames);
+    // Every jsonObject key must match a field name. If not, this check will fail. 
+    if (list.size() != jsonObject.keySet().size()) {
```

Comment:
I'm wondering if you can simply use the `containsAll` method to simplify the `addAll`, `retainAll`, size-check combination. For example, would `fieldNames.containsAll(jsonObject.keySet())` work in the first check?

Reply:
I think that would work in this case. Here I'm trying to check that every `jsonObject` key matches a `fieldName`, and I think `containsAll` would do that in a cleaner and more obvious way.